docs: deprecation workflow (#1067)
faker-js/faker: CONTRIBUTING.md

After adding new or updating existing locale data, you need to run `pnpm run generate:locales` to generate/update the related files.

## Deprecation workflow

If you ever find yourself deprecating some parts in the source code you can follow this steps to save yourself (and the reviewers) some trouble.

> If the code you want to deprecate is a property, convert it to a [getter](https://www.typescriptlang.org/docs/handbook/2/classes.html#getters--setters) first. Now that you have a function the first thing you want to do is call the internal [`deprecated` function](src\internal\deprecated.ts). Afterwars add a `@deprecated` parameter to the end of the JSDoc with a description of what to use instead.

Reviewer: ```suggestion
If the code you want to deprecate is a property, convert it to a [getter](https://www.typescriptlang.org/docs/handbook/2/classes.html#getters--setters) first. Now that you have a function the first thing you want to do is call the internal [`deprecated` function](src/internal/deprecated.ts). Afterwards add a `@deprecated` parameter to the end of the JSDoc with a description of what to use instead.
```

> If the code you want to deprecate is a property, convert it to a [getter](https://www.typescriptlang.org/docs/handbook/2/classes.html#getters--setters) first. Now that you have a function the first thing you want to do is call the internal [`deprecated` function](src/internal/deprecated.ts). Afterwards add a `@deprecated` parameter to the end of the JSDoc with a description of what to use instead.

Reviewer 2: Just a reminder for now:
Add `@see faker.module.method` if there exists an replacement method. It will be used to generate a link in our api docs.

Author: So I should not include this in the text?

Reviewer 2: Its part of the current workflow. Not sure if you want to change it though.
See also: https://github.com/faker-js/faker/issues/976

Reviewer: Don't know if this following thought is helpful or not, but anyways: I would say `@see` => links to the replacement, `@deprecation message` => message = a human readable text about what's going on 🤔

Reviewer 2: So lets go for both for now.

Author: I provided a sample.

Example:

```ts
/**
 * @deprecated Use your own implementation instead.
 */
get cat() {
  deprecated({

>     deprecated: 'faker.animal.cat()',

Reviewer 2: ```suggestion
    deprecated: 'faker.animal.cat',
```

After adding new or updating existing locale data, you need to run `pnpm run generate:locales` to generate/update the related files.

## Deprecation workflow

> If you ever find yourself deprecating some parts in the source code you can follow this steps to save yourself (and the reviewers) some trouble.

Reviewer 3: ```suggestion
If you ever find yourself deprecating something in the source code, you can follow these steps to save yourself (and the reviewers) some trouble.
```

> If the code you want to deprecate is a property, convert it to a [getter](https://www.typescriptlang.org/docs/handbook/2/classes.html#getters--setters) first. Now that you have a function the first thing you want to do is call the internal [`deprecated` function](src/internal/deprecated.ts). Afterwards add a `@deprecated` parameter to the end of the JSDoc with human readable description message on how to replace current implementations. Lastly add a `@see` parameter to the JSDoc with a link to the replacement in the faker library if it exist. The syntax for the link is `faker.[module].[function]`.

Reviewer 3: ```suggestion
If the code you want to deprecate is a property, convert it to a [getter](https://www.typescriptlang.org/docs/handbook/2/classes.html#getters--setters) first. Now that you have a function, the first thing you want to do is call the internal [`deprecated` function](src/internal/deprecated.ts). Afterwards, add a `@deprecated` parameter to the end of the JSDoc with a human readable description message with a suitable replacement for the deprecated function. Lastly, add a `@see` parameter to the JSDoc with a link to the replacement in the faker library (if it exists). The syntax for the link is `faker.[module].[function]`.
```
